crypto/src/e2e_identity/rotate.rs

```diff
@@ -280,24 +343,24 @@ impl MlsRotateBundle {
         use openmls::prelude::TlsSerializeTrait as _;
 
         let commits_size = self.commits.len();
-        let commits = self
-            .commits
-            .into_iter()
-            .try_fold(HashMap::with_capacity(commits_size), |mut acc, (id, c)| {
+        let commits = self.commits.into_iter().try_fold(
+            HashMap::with_capacity(commits_size),
+            |mut acc, (id, c)| -> Result<HashMap<String, MlsCommitBundle>> {
                 // because uniffi ONLY supports HashMap<String, T>
                 let id = hex::encode(id);
                 let _ = acc.insert(id, c);
-                CryptoResult::Ok(acc)
-            })?;
+                Ok(acc)
+            },
+        )?;
 
         let kp_size = self.new_key_packages.len();
-        let new_key_packages =
-            self.new_key_packages
-                .into_iter()
-                .try_fold(Vec::with_capacity(kp_size), |mut acc, kp| {
-                    acc.push(kp.tls_serialize_detached().map_err(MlsError::from)?);
-                    CryptoResult::Ok(acc)
-                })?;
+        let new_key_packages = self.new_key_packages.into_iter().try_fold(
+            Vec::with_capacity(kp_size),
+            |mut acc, kp| -> Result<Vec<Vec<u8>>> {
+                acc.push(kp.tls_serialize_detached()?);
+                Ok(acc)
+            },
+        )?;
```

Comment:
Just because we're touching this code anyway, if you want, let's refactor this to a `map()`/`collect()

Reply:
The reason I left these was the `with_capacity` calls. The [`Vec` documentation](https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#impl-FromIterator%3CT%3E-for-Vec%3CT%3E) states it clearly:

> Vec may use any or none of the following strategies, depending on the supplied iterator:
> 
> - preallocate based on [Iterator::size_hint()](https://doc.rust-lang.org/stable/std/iter/trait.Iterator.html#method.size_hint)

So it's not quite as simple as `map`/`collect`, unfortunately. Needs to be a `with_capacity`/`map`/`extend` setup to retain the existing performance characteristics, and then a wrapper so that we don't leave it mutable after the fact. In the end, I decided that retaining `try_fold` here was the simplest option.

Reply:
Good to see a use case where folding to a collection is a valid approach.